pkg/operator/ceph/object/spec.go

```diff
@@ -337,6 +337,9 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 		return v1.Container{}, errors.Wrap(err, "failed to generate default readiness probe")
 	}
 
+	clientID := strings.TrimPrefix(generateCephXUser(rgwConfig.ResourceName), "client.")
+	daemonID := fmt.Sprintf("%s.$(POD_UID)", clientID)
```

Comment:
The error you're getting looks like it's trying to use the cephx user wtih the pod_uid suffix to connect. However, the cephx key created for all the daemons to connect with is generated without the pod_uid:
https://github.com/rook/rook/blob/786b7555947649a10e4299196a6f936712a23c9e/pkg/operator/ceph/object/config.go#L98-L111

So we need to find a way to use the same keyring for all the daemons, but the different name with the pod_uid suffix for the admin port

Reply:
This must mean that cephadm is using a different keyring for each rgw daemon?